src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/UpdateContactDetailsService.java

```diff
@@ -0,0 +1,121 @@
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_PHONE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_RESIDE_OUTSIDE_UK;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_ADDRESS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_DX_NUMBER;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_FIRM;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_NAME;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_NOTIFICATIONS_EMAIL_CONSENT;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_PHONE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_PHONE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.SOLICITOR_REFERENCE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.service.CaseDataService.nullToEmpty;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UpdateContactDetailsService {
+
+    public void persistOrgPolicies(Map<String, Object> caseData, CaseDetails originalDetails) {
+        caseData.put(APPLICANT_ORGANISATION_POLICY, originalDetails.getData().get(APPLICANT_ORGANISATION_POLICY));
+        caseData.put(RESPONDENT_ORGANISATION_POLICY, originalDetails.getData().get(RESPONDENT_ORGANISATION_POLICY));
+    }
+
+    public boolean isIncludesRepresentationChange(Map<String, Object> caseData) {
+        return YES_VALUE.equals(caseData.get(INCLUDES_REPRESENTATION_CHANGE));
+    }
+
+    public void handleApplicantRepresentationChange(CaseDetails caseDetails) {
+        String caseTypeId = caseDetails.getCaseTypeId();
+        Map<String, Object> caseData = caseDetails.getData();
+
+        String nocParty = (String) caseData.get(NOC_PARTY);
+        if (APPLICANT.equalsIgnoreCase(nocParty)) {
+            removeApplicantSolicitorDetails(caseData, caseTypeId);
+        }
+    }
+
+    public void handleRespondentRepresentationChange(CaseDetails caseDetails) {
+        String caseTypeId = caseDetails.getCaseTypeId();
+        Map<String, Object> caseData = caseDetails.getData();
+
+        String nocParty = (String) caseData.get(NOC_PARTY);
+        if (RESPONDENT.equalsIgnoreCase(nocParty)) {
+            removeRespondentDetails(caseData, caseTypeId);
+        }
+    }
+
+    static void removeApplicantSolicitorDetails(Map<String, Object> caseData, String caseTypeId) {
```

Comment:
Why is this method static?

Reply:
[28bf076](https://github.com/hmcts/finrem-case-orchestration-service/pull/1860/commits/28bf07620e8ee92e0bf95a2e6dd28b118707f118)